tendermint/tests/lite.rs

```diff
@@ -119,6 +119,8 @@ impl MockRequester {
     }
 }
 
+// Link to the commit that generated below JSON test files:
+// https://github.com/Shivani912/tendermint/commit/f7d16ab59b55a4f1a5cdbfa6b0c24467aa88fdb2
```

Comment:
@Liamsi Here is the link to the commit that generated the new val_set_tests.json

Reply:
I added this test that fails with FaultyFullNode error:
https://github.com/interchainio/tendermint-rs/blob/b28b3da6049cb4b0868e3c234f447c22daa2d289/tendermint/tests/support/lite/single_step_sequential/val_set_tests.json#L13078

Here, the input contains a validator set of 3 vals (v1, v2, v3) for header at height 2 with precommits from 3 vals (v1, v2, v4). This makes it pass all other validation tests but fails because there is a vote from val v4 which is not in the val set and thus there is a chance that the full node who sent this data is misbehaving. 

@Liamsi